### PR TITLE
Adding Screen space shadows in HDRP

### PR DESCRIPTION
**Pull request by anisunity**
Adding screen space shadows settings to the render pipeline asset
Refactoring the HDRaytracingShadowManager to a ScreenSpaceShadowManager that also includes rasterized shadows
Displaying both RT and Rasterized parameters when having ray traced area light shadows
Adding a use screen space property to the hdadditionallightdata
Renaming the raytraced area shadow debug display to screen space shadows and supporting multiple shadows debug
Splitting the notion of "WillRenderShadows" to "WillRenderShadowMap" "WillRenderScreenSpaceShadow" and "WillRenderRayTracedShadows"
Adding a screenspaceshadow in the directionallightdata
Falling back on shadowmap for raytraced area shadow on transparent surfaces
supporting XR for screen space shadows
converting ScreenSpaceShadowManager to an hdrp partial class
Adding a framesettings for screen space shadows

https://katana.bf.unity3d.com/projects/com.unity.render-pipelines/builders?ScriptableRenderLoop_branch=HDRP%2FRayTracedDirectionalShadows&automation-tools_branch=master&unity_branch=trunk#

### CODE REVIEW

File: com.unity.render-pipelines.high-definition/Runtime/Lighting/LightLoop/LightLoop.cs [hunk 1]
```diff
@@ -927,7 +924,7 @@ public Vector3 GetLightColor(VisibleLight light)
             return new Vector3(light.finalColor.r, light.finalColor.g, light.finalColor.b);
         }
 
-        public bool GetDirectionalLightData(CommandBuffer cmd, GPULightType gpuLightType, VisibleLight light, Light lightComponent, HDAdditionalLightData additionalLightData, AdditionalShadowData additionalShadowData, int lightIndex, int shadowIndex, DebugDisplaySettings debugDisplaySettings, int sortedIndex)
+        public bool GetDirectionalLightData(CommandBuffer cmd, GPULightType gpuLightType, VisibleLight light, Light lightComponent, HDAdditionalLightData additionalLightData, AdditionalShadowData additionalShadowData, int lightIndex, int shadowIndex, ref int screenSpaceShadowIndex, DebugDisplaySettings debugDisplaySettings, int sortedIndex)
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Runtime/Lighting/LightLoop/LightLoop.cs]: put the ref at the end of the function.

File: com.unity.render-pipelines.high-definition/Runtime/Lighting/Shadow/ScreenSpaceShadowManager.cs [hunk 2]
```diff
@@ -97,23 +87,49 @@ static RTHandleSystem.RTHandle AreaAnalyticHistoryBufferAllocatorFunction(string
...
+            {
+                HDUtils.SetRenderTarget(cmd, m_ScreenSpaceShadowTextureArray, depthSlice: m_CurrentSunLightDirectionalLightData.screenSpaceShadowIndex);
+                HDUtils.DrawFullScreen(cmd, s_ScreenSpaceShadowsMat, m_ScreenSpaceShadowTextureArray);
+            }
+        }
+        public bool RenderAreaShadows(HDCamera hdCamera, CommandBuffer cmd, int frameCount)
+        {
             // Let's check all the resources and states to see if we should render the effect
-            HDRaytracingEnvironment rtEnvironment = m_RaytracingManager.CurrentEnvironment();
+            HDRaytracingEnvironment rtEnvironement = m_RayTracingManager.CurrentEnvironment();
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Runtime/Lighting/Shadow/ScreenSpaceShadowManager.cs]: "rtEnvironment ", you have revert the correction :)

Comment (sebastienlagarde) [reply com.unity.render-pipelines.high-definition/Runtime/Lighting/Shadow/ScreenSpaceShadowManager.cs]: m_RaytracingManager

Comment (anisunity) [reply com.unity.render-pipelines.high-definition/Runtime/Lighting/Shadow/ScreenSpaceShadowManager.cs]: Let's discuss the Ray Tracing/raytracing part

**Comment by sebastienlagarde** [comment]
This PR have been merged in to https://github.com/Unity-Technologies/ScriptableRenderPipeline/pull/3776

**Pull request closed without merging**